moler/device/atremote.py

```diff
@@ -0,0 +1,225 @@
+# -*- coding: utf-8 -*-
+"""
+Moler's device has 2 main responsibilities:
+- be the factory that returns commands of that device
+- be the state machine that controls which commands may run in given state
+"""
+
+__author__ = 'Grzegorz Latuszek'
+__copyright__ = 'Copyright (C) 2020, Nokia'
+__email__ = 'grzegorz.latuszek@nokia.com'
+
+from moler.device.textualdevice import TextualDevice
+from moler.device.proxy_pc import ProxyPc
+from moler.device.unixlocal import UnixLocal
+from moler.device.unixremote import UnixRemote
+from moler.helpers import call_base_class_method_with_same_name, mark_to_call_base_class_method_with_same_name
+from moler.cmd.at.genericat import GenericAtCommand
+
+
+@call_base_class_method_with_same_name
+class AtRemote(UnixRemote):
+    r"""
+    AtRemote device class.
+
+    Example of device in yaml configuration file:
+    -without PROXY_PC:
+      AT_1:
+       DEVICE_CLASS: moler.device.atremote.AtRemote
+       CONNECTION_HOPS:
+         UNIX_LOCAL:
+           UNIX_REMOTE:
+             execute_command: ssh # default value
+             command_params:
+               expected_prompt: unix_remote_prompt
+               host: host_ip
+               login: login
+               password: password
+      UNIX_REMOTE:
+        AT_REMOTE:
+          execute_command: run_serial_proxy # default value
+          command_params:
+            serial_devname: 'COM5'
+      AT_REMOTE:
+        UNIX_REMOTE:
+          execute_command: exit_serial_proxy # default value
+    """
+
+    at_remote = "AT_REMOTE"
+
+    def __init__(self, sm_params, name=None, io_connection=None, io_type=None, variant=None, io_constructor_kwargs={},
```

Comment:
io_constructor_kwargs = None because default value may be changed (not empty dict) )during run time.